tabby-ssh/src/components/sftpPanel.component.ts

```diff
@@ -106,17 +109,23 @@ export class SFTPPanelComponent {
         }
     }
 
+    async openCreateDirectoryModal (): Promise<void> {
+        const modal = this.ngbModal.open(SFTPCreateDirectoryModalComponent)
+        const directoryName = await modal.result
+        if (directoryName !== '') {
+            this.sftp.mkdir(path.join(this.path, directoryName)).finally(() => {
+                this.navigate(path.join(this.path, directoryName))
+            })
+        }
+    }
+
     async upload (): Promise<void> {
         const transfers = await this.platform.startUpload({ multiple: true })
         await Promise.all(transfers.map(t => this.uploadOne(t)))
     }
 
     async uploadOne (transfer: FileUpload): Promise<void> {
         await this.sftp.upload(path.join(this.path, transfer.getName()), transfer)
-        const savedPath = this.path
-        if (this.path === savedPath) {
-            await this.navigate(this.path)
-        }
```

Comment:
You've correctly noticed that this is wrong! I've just fixed the statement order in the master branch - you'll need to merge it back into your branch.